ssl/t1_lib.c

```diff
@@ -1692,7 +1693,8 @@ unsigned char *ssl_add_serverhello_tlsext(SSL *s, unsigned char *buf,
         s2n(0, ret);
     }
 
-    if (s->s3->alpn_selected) {
+    /* send only if proposed to us and selected */
```

Comment:
How can it happen here that `alpn_selected` is set but `alpn_proposed` is not?

Also, style nit: explicit pointer checks, please: `s->s3->alpn_selected != NULL`.